Add item: Button, modal, and styling

codecaviette opened the pull request:
## Description

When a user is on the List page, they will see a button to add a new item. If they have items on their list, the button will be located under their list. When they click on the Add Item button, they'll be presented with a modal to add their item and select when to purchase the item next. When the Add Item button is tapped, the user will see one of the following three Alerts: 
- Success: If the user has submitted a new item to their list successfully, they will see a green alert saying "Item successfully added to your list!"
- Failed - Empty field/s: If either of the fields are not filled out, the user will see a red alert saying "Item was NOT added because one or more fields were incomplete. Please fill out all fields."
- Failed - Duplicate item: If the user attempts to submit a duplicate item to the list, they'll see an orange alert with the message "You've already added this item, so we were not able to add it to your list." 

Note that currently, the font sizes are a bit off across the app. As a team, we will need to decide on how to build in consistency across the website, maybe setting a global font size. 
 

## Related Issue
closes 18


## Acceptance criteria
- [x] Create and style Add Item button
- [x] Create and style Add Item modal 
- [x] Add styling for drop-down menu to select when to next purchase item

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
| ✓ | Add styling              |

## Updates

### Before

<img width="1137" alt="Screenshot 2023-10-03 at 4 23 15 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/7c87c81a-9548-4d9e-bf48-7225af2fdadd">


### After

Add Item button
<img width="704" alt="Screenshot 2023-10-03 at 6 34 26 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/7ba1aae1-a933-409e-959b-75c7707f24a3">

Add Item modal
<img width="704" alt="Screenshot 2023-10-05 at 12 18 10 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/a44a3e69-c03e-4638-9799-bde29c879ada">

Alert after successful submission
<img width="704" alt="Screenshot 2023-10-03 at 6 34 54 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/a5d3889a-ea2e-4423-8089-8a4e345fd09c">

Alert after failed submission
<img width="704" alt="Screenshot 2023-10-03 at 6 35 04 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/63386589/8d2b75b1-e258-4c78-a67b-bfd1299be0ee">




## Testing Steps / QA Criteria

1. Open the main branch in your terminal and pull all remote changes with `git pull`
2. Navigate to this feature branch with command `git switch modal-add-item`
3. Run `npm start` to spin up the feature in the dev web environment.
4. In your browser, navigate to the List page of the dev website.
5. If you need to create a list, go ahead and do so. 
6. On the List page, tap the Add Item button, which will show you a modal. Both fields should be outlined in red indicating that it needs to be updated. Add a new item and select a time frame from the drop-down menu. The red outlines should no longer show. Click the Add Item button.
7. The modal should automatically close, and you should see a green alert saying "Item successfully added to your list!"
8. Go to the Firestore database to confirm your item was added to your list.
9. Navigate back to the List page and click the Add Item button. This time, leave either or both of the fields untouched and tap the Add Item button. 
10. The modal should automatically close, and you should see a red alert saying "Item was NOT added because one or more fields were incomplete. Please fill out all fields."
11. Navigate back to the List page and click the Add Item button. Now this time, enter a duplicated item, select a timeframe from the drop-down menu, and tap the Add Item button. 
12. The modal should automatically close, and you should see an orange alert with the message "You've already added this item, so we were not able to add it to your list."

Commit:
Adds a add item button and modal with form to list view

Commit:
Updates add item modal, fixing submit functionality. Adds alerts for form success and errors

Commit:
Updates logo

codecaviette committed:
Add styling to Add Item button and modal

lyleschemmerling commented:
We may want to consider center-aligning the list,, and I would also maybe not hight the fields of the add item form as an error until the user tries to submit without a name

MagdaSlifierz commented on src/components/AddItem.jsx:
```diff
@@ -0,0 +1,184 @@
+						</form>
+					</ModalBody>
+
+					<ModalFooter>
+						<Button
+							leftIcon={<AddIcon />}
+							form="addItemForm"
+							type="submit"
+							onClick={onClose}
+							bg="brand.navy"
+							color="brand.off_white"
+						>
+							Add Item
+						</Button>
+					</ModalFooter>
```
I recommend considering a slight modification to the button's visibility. Currently, when hovered upon, it turns white, making it difficult to discern the text "Add Item."

As you mentioned in your note, it's essential for all of us to reach a consensus on the button's size and styling.

<img width="535" alt="Screenshot 2023-10-04 at 6 47 13 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/49603115/783f15e4-b790-4142-bebc-c516590abe2b">

MagdaSlifierz reviewed (commented):
I've also observed that when we click on the "Add Item" link in the navbar, we're redirected to a page with an "Add Item" button that doesn't match the styling. Furthermore, when the form appears, it exhibits a different style compared to the one used on the list page.
<img width="1428" alt="Screenshot 2023-10-04 at 7 09 26 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/49603115/09ec7551-658a-49ec-a733-06c74fc02122">
<img width="1419" alt="Screenshot 2023-10-04 at 7 09 50 PM" src="https://github.com/the-collab-lab/tcl-62-smart-shopping-list/assets/49603115/ba0a2b57-ec01-4686-9744-b2c3d1e956da">

MagdaSlifierz commented:
I'd like to share a thought that just crossed my mind. Perhaps we could consider making the "Add Item" button/link in the navbar completely invisible. This way, we wouldn't need to style this particular page with just a single button. We could maintain the functionality, where when a new list is created, the user is directed to a page with no items and the "Add Item" button. Another scenario: when the user already has a token/ list, the "Add Item" button is placed right after the list.

codecaviette committed:
Removed requirement of modal input fields, Added more color contrast to modal input fields and Add Item button

codecaviette commented:
Thanks for reviewing, Lyle and Magda! 

@lyleschemmerling: Agreed the items should be centered, and this will be included in the styling of the List page in Issue 17. Also agree that the red highlight of the input fields is a bit premature-- I've removed that functionality since the user is notified of any errors after they tap the Add Item button in the modal. 

@MagdaSlifierz: I've updated the hover styling of the Add Item button in the modal so that it's easier to read when hovering over it-- good catch! As for the Add Item tab/page, I thought we'd discussed as a team hiding it entirely and opting for this Add Item modal instead, which is why navigating to the actual page looks fairly janky.

avinashi10 reviewed (commented):
- Regarding Lyle's comment, if the choice is between having the red error fields immediately appear vs removing them all together and the form lets you submit even without required fields filled out, I would say for UI the first choice is a lot better. It's a lot more intuitive that the submit button won't work if you don't have required fields filled out than it seeming like you added an item and having to notice an alert. Jenny, I know we agreed that an alert dialog (that pops up) is a stretch goal. And we I believe it was not straightforward to keep the fields required but only have the red show up on submit if the field is empty. 

- With the alert message that appears, they all appear at the bottom of the screen and if you test with a long list (e.g. date mean hide) you have to scroll down to see the alert so when I added an empty item or a duplicate it seemed like the add item function wasn't working and it took me a second to scroll down to find the alert (and that was only bc I knew to look for it). We should move the alerts to the top of the list - but that does add a layer of complexity bc they would need to be a separate component and we would need to pass the 3 state props that conditionally render them to that component.

- The styling looks great and the spacing is improved too! Is there a way to make the heading 'Add an item to your list' have a different background color (and text color if needed)? I think we should also add a lighter color to our custom theme for the background. That 'off_white' looks very yellow/cream-colored on my screen.

- Regarding the add item navigation tab that leads to the separate page - we did agree to remove that. Though thinking about the UI - it might be nice to have it up there because with long lists (like I mentioned above) the add item button is not obvious. We can either try to have the add item navigation also render the modal component or we have place the add item button at a fixed place on the screen rather than at the bottom of the list so that it's always on the screen view and the user doesn't need to scroll to get to it.

Happy to work on all of these together!

codecaviette committed:
Move Add Item button to top of List page to ensure visibility; Hide Add Item page

MagdaSlifierz reviewed (approved):
Lgtm, that was a good idea to hide the Add Item button from the navbar. 👍

codecaviette committed:
Resolved merge conflict

codecaviette committed:
Move button above list

avinashi10 reviewed (approved):
looks good! I'm confused why github says there are no merge conflicts with main as I'm looking at Layout.jsx and it looks quite different here than in main
